x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfigSerializingTests.java

```diff
@@ -135,11 +120,43 @@ public void testValidateMatchingField() throws IOException {
         when(fieldCaps.isAggregatable()).thenReturn(true);
         responseMap.put("my_field", Collections.singletonMap("date", fieldCaps));
 
-        DateHistoGroupConfig config = new DateHistoGroupConfig.Builder()
-                .setField("my_field")
-                .setInterval(new DateHistogramInterval("1d"))
-                .build();
+        DateHistogramGroupConfig config = new DateHistogramGroupConfig("my_field", new DateHistogramInterval("1d"), null, null);
         config.validateMappings(responseMap, e);
         assertThat(e.validationErrors().size(), equalTo(0));
     }
+
+    public void testBwcSerialization() throws IOException {
```

Comment:
Isn't this sort of test covered by `AbstractSerializingTestCase` via it's super-class `AbstractWireTestCase`?  Or am I missing something that makes this different?

Reply:
It's a bit different: `AbstractSerializingTestCase` takes care of testing the XContent parsing logic and  `AbstractWireTestCase` tests the java serialization logic. But this pull request changes the format of the timezone from `DateTimeZone` to `String` and `AbstractWireTestCase` will now execute the tests using the new type only.

This test checks that we can still read from / write to the old format that serializes/deserializes DateTimeZone objects.

Reply:
Ooh, missed that little detail.  Great thanks!